src/main/java/teammates/ui/template/FeedbackSessionPublishButton.java

```diff
@@ -0,0 +1,83 @@
+package teammates.ui.template;
+
+import teammates.common.datatransfer.FeedbackSessionAttributes;
+import teammates.common.datatransfer.InstructorAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.Sanitizer;
+import teammates.ui.controller.PageData;
+
+public class FeedbackSessionPublishButton {
+
+    private boolean hasUnpublish;
+    private boolean hasPublish;
+
+    private String tooltipText;
+
+    private String unpublishLink;
+    private String publishLink;
+
+    private boolean allowedToUnpublish;
+    private boolean allowedToPublish;
+
+    private String toggleUnpublishSessionParams;
+    private String togglePublishSessionParams;
+    
+    public FeedbackSessionPublishButton(PageData data, FeedbackSessionAttributes session,
+                                        boolean isHome, InstructorAttributes instructor) {
+        String courseId = session.courseId;
+        String feedbackSessionName = session.feedbackSessionName;
+
+        this.hasUnpublish = !session.isWaitingToOpen() && session.isPublished();
+        this.hasPublish = !session.isWaitingToOpen() && !session.isPublished();
```

Comment:
Let me look more into this, might warrant opening an issue to improve readability and maintainability, perhaps I can screenshot the code and open so that we can have a discussion about it and more people might look at it.